[feature] show popup for version update

codingCoffee opened the pull request:
- added a new weekly hook to check if new version update is available
- added function to desk.js to show popup in case a new version is
available
- added redis wrappers for set related commands, namely sadd, srem,
sismember, spop, srandmember, smembers

**Edit: Screenshot updated in a comment below!**

Screenshot:
![screenshot_20180812_025457](https://user-images.githubusercontent.com/13611153/43996194-1a86589e-9ddb-11e8-83fb-d0552d1a243d.png)

Please read the [Pull Request Checklist](https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist) to ensure you have everything that is needed to get your contribution merged.

codingCoffee commented:
Sidenote: The codacy warning can be ignored. I'm not piping any user input to the shell.

rmehta commented:
@codingCoffee can we the number as well too? And a link to the changelog?

Also how does this read if multiple apps have updates?

codingCoffee committed:
[feature] show popup for version update

- added a new weekly hook to check if new version update is available
- added function to desk.js to show popup in case a new version is
available
- added redis wrappers for set related commands, namely sadd, srem,
sismember, spop, srandmember, smembers

codingCoffee commented:
**Functionality**:
Clicking on the app takes you to the GitHub release page so for instance for frappe v10.1.48 it'll take you [here](https://github.com/frappe/erpnext/releases/tag/v10.1.48)

**In case of multi type releases:**
![screenshot_20180816_014549](https://user-images.githubusercontent.com/13611153/44170872-4bdd6680-a0f6-11e8-8550-de4bdf6db7ba.png)

**In case of single type releases**
![screenshot_20180816_014520](https://user-images.githubusercontent.com/13611153/44170786-091b8e80-a0f6-11e8-9c2c-32998acfee3b.png)

felixvarghese commented:
By default, can this only be shown for users with the "System Manager" role rather than everyone?

Otherwise, what will likely happen is people are just going to comment out the weekly hook, and that will end up being completely counterproductive to the intended goal.

codingCoffee committed:
[minor] pop up shown to system managers and functionified code

codingCoffee commented:
Made the necessary modifications!

rmehta commented on frappe/utils/change_log.py:
```diff
@@ -124,3 +125,85 @@ def get_app_last_commit_ref(app):
+	# "update-user-set" will be a set of users
+
+def add_message_to_redis(update_message):
+	update_message += "<b>Please ask your system manager to update your instance</b>"
```
no longer required!

chdecultot commented on frappe/utils/change_log.py:
```diff
@@ -124,3 +125,85 @@ def get_app_last_commit_ref(app):
+				title             = app.title
+			)
+		if release_links:
+			update_message += "New {} releases for the following apps are available:<br><br>{}<hr>".format(update_type, release_links)
```
Hi @codingCoffee,

Sorry for the comment, but is it possible to make this message translatable for non-english speakers please ?

I guess the update type should also be translatable to fit in.

Thanks a lot!

codingCoffee replied:
@chdecultot thanks a lot for the feedback :D
Amend the code accordingly!

codingCoffee committed:
translated strings and removed admin upgrade message

rmehta commented on frappe/utils/change_log.py:
```diff
@@ -124,3 +126,84 @@ def get_app_last_commit_ref(app):
+				title             = app.title
+			)
+		if release_links:
+			update_message += _("New {} releases for the following apps are available") + ":<br><br>{}<hr>".format(update_type, release_links)
```
Not a good idea to encode the translated message (if the users have different languages)?

Better to just save the release JSONs and build the message at the point of display.

netchampfaris committed:
Add only System Managers to update-user-set

codingCoffee committed:
update-info: put json in redis, and generate message at user end
